testing-modules/rest-assured/src/test/java/com/baeldung/restassured/RestAssuredMultipartIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+package com.baeldung.restassured;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aMultipart;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.configureFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.containing;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.springframework.http.HttpStatus.OK;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.matching.MultipartValuePatternBuilder;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.MultiPartSpecBuilder;
+import io.restassured.specification.MultiPartSpecification;
+
+class RestAssuredMultipartIntegrationTest {
+
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String FILE = "file";
+    private static final String FILENAME1 = "baeldung.txt";
+    private static final String FILENAME2 = "BAELDUNG.txt";
+    private static final String MULTIPART_FORM_DATA = "multipart/form-data";
+    private static final String UPLOAD_URL = "/upload";
+
+    private WireMockServer wireMockServer;
+
+    @BeforeEach
+    void startServer() {
+        final int port = Util.getAvailablePort();
+        wireMockServer = new WireMockServer(port);
+        wireMockServer.start();
+        configureFor("localhost", port);
+        RestAssured.port = port;
+    }
+
+    @AfterEach
+    void stopServer() {
+        wireMockServer.stop();
+    }
+
+    @Test
+    void whenUploadOneFile_ThenSuccess() {
+        stubFor(post(urlEqualTo(UPLOAD_URL)).withHeader(CONTENT_TYPE_HEADER, containing(MULTIPART_FORM_DATA))
+            .withRequestBody(containing(FILE))
+            .withRequestBody(containing(getFileContent(FILENAME1)))
+            .willReturn(aResponse().withStatus(OK.value())));
+
+        given().multiPart(FILE, getFile(FILENAME1))

```

Comment:
If this test method `throws Exception` then the test will fail if the `getFile` fails.